### PR TITLE
ceph-: abitlity to copy admin on all the nodes

### PR DESCRIPTION
**Pull request by leseb**
This commit allows you to set a new variable to 'true' if you want to
have ceph admin key copied over different kind of hosts such as MDS,
OSD, RGW. To enable this just set `copy_admin_key` to true.

Closes: #555

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
Hum somehow, the key always gets copied even if set to False and evaluated as False.

@andrewschoen @mattbenjamin @alfredodeza any idea?

File: roles/ceph-osd/tasks/pre_requisite.yml [hunk 1]
```diff
@@ -23,10 +23,14 @@
...
     owner: "{{ key_owner }}"
     group: "{{ key_group }}"
     mode: "{{ key_mode }}"
+  with_items:
+    - { name: /var/lib/ceph/bootstrap-osd/ceph.keyring, copy: true }
+    - { name: /etc/ceph/ceph.client.admin.keyring, copy: "{{ copy_admin_key }}" }
   when:
-    cephx
+    cephx and
+    item.copy
```
Comment (mattt416) [inline roles/ceph-osd/tasks/pre_requisite.yml]: The only way I could get this to work was to do this instead:

```
when:
  cephx and
  item.copy | bool
```

**Pull request merged**